app/src/main/java/com/example/mycalculator/MainActivity.java

```diff
@@ -2,13 +2,99 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.annotation.SuppressLint;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.example.mycalculator.databinding.ActivityMainBinding;
+import com.example.mycalculator.mvp.CalculatorModelImpl;
+import com.example.mycalculator.mvp.CalculatorPresenterImpl;
+import com.example.mycalculator.mvp.CalculatorViewImpl;
 
 public class MainActivity extends AppCompatActivity {
 
+
+    private ActivityMainBinding binding;
+    private CalculatorPresenterImpl presenter;
+
+
+    @SuppressLint("WrongViewCast")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new CalculatorPresenterImpl(new CalculatorModelImpl(), new CalculatorViewImpl(this, binding));
+
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed("0");
```

Comment:
pls remove the hardcode, you can move this to a Util class, create a class with all constant.